### ISSUE TITLE
startup.m2.in inconsistencies between autotools and cmake builds

### ISSUE THREAD

**Issue by d-torrance** [Macaulay2/M2#2780]
There are some differences between how `startup.m2.in` is generated and its file location information is stored between the two build systems.

### autotools

In the file `Macaulay2/d/startup.c`:

```c
cached_file startupFile = {
"Macaulay2/Core/startup.m2.in"
```

And an error message:

```m2
profzoom@peg-amy:~/src/macaulay2/M2/M2/BUILD/doug$ usr-dist/x86_64-Linux-Ubuntu-22.04/bin/M2 --check 7
Macaulay2/Core/startup.m2.in:562:33:(0):[1]: --check 7: expected 1, 2, or 3
Macaulay2/Core/startup.m2.in:562:33:(0):[1]: --entering debugger (type help to see debugger commands)
```

### cmake

In the file `Macaulay2/bin/startup.c` (note the directory):

```c
cached_file startupFile =
  {
   "/home/profzoom/src/macaulay2/M2/M2/Macaulay2/m2/startup.m2.in",
```

And an error message:

```m2
profzoom@peg-amy:~/src/macaulay2/M2/M2/BUILD/doug-cmake$ usr-dist/x86_64-Linux-Ubuntu-22.04/bin/M2 --check 7
../../Macaulay2/m2/startup.m2.in:562:33:(0):[1]: --check 7: expected 1, 2, or 3
../../Macaulay2/m2/startup.m2.in:562:33:(0):[1]: --entering debugger (type help to see debugger commands)
```

I have no strong opinion about which is better, but they should be consistent.

**Comment by mahrud** [comment]
I don't think consistency is necessarily a virtue outside of mathematical things. For instance the brew build or the debian builds have many small differences.

In this case, the cmake build behaves that way to make development easier by pointing to the file which needs to be changed and recompiled to correct the error.

**Comment by d-torrance** [comment]
Now that I've thought about it a little more, I do like the cmake behavior better for development.  But for distribution, the source directory wouldn't be available, or even if it was, it would almost certainly have a different path.  In those cases, it would probably be good to have it point to `${datadir}/Macaulay2/Core/startup.m2`.  Even though any changes the user might make wouldn't do anything, it would at least give them a chance to see the code that's responsible for the error.  (The autotools behavior is pretty pointless.)

There's a `DEVELOPMENT` macro that seems like it might be good to differentiate between these cases, but it also inserts `-Werror` into the build flags, which causes the build to fail because of #2274.

**Comment by mahrud** [comment]
I'd rather not change the behavior of cmake. Errors in startup.m2 are exceedingly rare, so I'm not sure if it's worth complicating the build system for this.

Perhaps a better solution would be to figure out a way not to hardcode that file at all.

**Comment by d-torrance** [comment]
> Perhaps a better solution would be to figure out a way not to hardcode that file at all.

Yes, that definitely would be the best solution!  Why are things the way they are now?  If `startup.m2` could just be a regular old .m2 file in the `Core` directory, none of this would be an issue.

**Comment by mahrud** [comment]
My understanding of the problem is that M2-binary doesn't know where to look for startup.m2 to begin with.

**Comment by d-torrance** [comment]
Closing -- this was fixed in #2781.
